midterm-2023-11-06/code/5175-kenan-sipahi/script.ts

```diff
@@ -0,0 +1,378 @@
+        sortCountrySwitch=true;
+    }
+    else if (sortCountrySwitch===true)
+    {
+        const sortedBands=bands.toSorted(countrySorter_desc);
+        displayBands(sortedBands);
+        sortCountrySwitch=false;
+    }
+
+   
+    
+}
+
+
+
+
+const sortAlbums =()=>
+{
+    if(sortAlbumsSwitch===false){
+        const sortedBands=bands.toSorted(albumsSorter_asc);
+        displayBands(sortedBands)   
+        sortAlbumsSwitch=true;
+
+    }
+    else if (sortAlbumsSwitch===true)
+    {
+        const sortedBands=bands.toSorted(albumsSorter_desc);
+        displayBands(sortedBands)   
+        sortAlbumsSwitch=false;
+    }
+   
+}
+
+
+
+const displayBands=(bands: MetalBand[]) =>{
+    const container = document.getElementById("band-container")!;
+    container.innerHTML="";
+    for (const band of bands){
+        const metalBandRow = generateRowBands(band);
+        container.append(metalBandRow);
+    }
+}
+
+const findCountry=(band:MetalBand)=>{
+        let bandlocation = band.location;
+        let parts = bandlocation.split(', ');
+        let lastPart = parts[parts.length - 1];
+        return lastPart;
+}
+const generateRowBands= (bands : MetalBand) =>{
+
+    const row = document.createElement("div");
+        row.classList.add("band-table");
+
+        const idCell = document.createElement("div");
+        idCell.classList.add("band-data","band-id");
+        idCell.innerHTML=bands.id.toString();
+        row.appendChild(idCell);
+
+        const nameCell = document.createElement("div");
+        nameCell.classList.add("band-data","band-name");
+        nameCell.innerHTML=bands.name;
+        row.appendChild(nameCell);
+
+        const genreCell = document.createElement("div");
+        genreCell.classList.add("band-data","band-genre");
+        genreCell.innerHTML=bands.genre;
+        row.appendChild(genreCell);
+
+        const formedCell = document.createElement("div");
+        formedCell.classList.add("band-data","band-formed");
+        formedCell.innerHTML=bands.formed.toString();
+        row.appendChild(formedCell);
+
+        
+        const locationCell = document.createElement("div");
+        locationCell.classList.add("band-data","band-location");
+        locationCell.innerHTML=bands.location;
+        row.appendChild(locationCell);
+
+        
+        const countryCell = document.createElement("div");
+        countryCell.classList.add("band-data","band-country");
+       
+      
+        countryCell.innerHTML=findCountry(bands);
+        row.appendChild(countryCell);
+
+   
+      
+    
+
+
+        const membersCell = document.createElement("div");
+        membersCell.classList.add("band-data","band-members");
+        membersCell.innerHTML=bands.members.toString();
+        row.appendChild(membersCell);
+
+        const albumsCell = document.createElement("div");
+        albumsCell.classList.add("band-data","band-albums");
+
+        const firstalbum=`${bands.albums[0].year} - "${bands.albums[0].name}"`
```

Comment:
This assumes that the albums are in a chronological order, when the data does not explicitly guarantees that.